spacy_llm/backends/integration/base.py

```diff
@@ -0,0 +1,22 @@
+import dataclasses
+from typing import Iterable, Generic, TypeVar, Callable
+
+# Object/Callable from third-party lib used to execute prompts.
+_Integration = TypeVar("_Integration")
+# Type of prompts returned from Task.generate_prompts().
+_PromptType = TypeVar("_PromptType")
+# Type of responses returned from query function.
+_ResponseType = TypeVar("_ResponseType")
```

Comment:
I don't really agree with `_PromptType` and `_ResponseType` being typevars. Like these are strs can't they just be strs?

Reply:
I'm not an expert of other backends, but we might want to support more exotic types in the future, right? For instance including richer agent logic within the prompt might be a legitimate use case?

Reply:
Yes, they can be any type.

Reply:
Ahh ok thanks for the background @bdura, makes sense